python/config_settings/config_settings.bzl

```diff
@@ -15,26 +15,58 @@
 """This module is used to construct the config settings in the BUILD file in this same package.
 """
 
+load("@bazel_skylib//lib:selects.bzl", "selects")
 load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
 
-# buildifier: disable=unnamed-macro
-def construct_config_settings(python_versions):
+def construct_config_settings(name, python_versions):
     """Constructs a set of configs for all Python versions.
 
     Args:
-        python_versions: The Python versions supported by rules_python.
+        name: str, unused; only specified to satisfy buildifier lint checks
+            and allow programatic modification of the target.
+        python_versions: list of all (x.y.z) Python versions supported by rules_python.
     """
+
+    # Maps e.g. "3.8" -> ["3.8.1", "3.8.2", etc]
+    minor_to_micro_versions = {}
+    allowed_flag_values = []
+    for micro_version in python_versions:
+        minor, _, _ = micro_version.rpartition(".")
+        minor_to_micro_versions.setdefault(minor, []).append(micro_version)
+        allowed_flag_values.append(micro_version)
+
     string_flag(
         name = "python_version",
+        # TODO: The default here should somehow match the MODULE config
         build_setting_default = python_versions[0],
-        values = python_versions,
+        values = sorted(allowed_flag_values),
         visibility = ["//visibility:public"],
     )
 
-    for python_version in python_versions:
-        python_version_constraint_setting = "is_python_" + python_version
+    for minor_version, micro_versions in minor_to_micro_versions.items():
+        # This matches the raw flag value, e.g. --///python/config_settings:python_version=3.8
+        # It's private because matching the concept of e.g. "3.8" value is done
+        # using the `is_python_X.Y` config setting group, which is aware of the
+        # minor versions that could match instead.
+        equals_minor_version_name = "_python_version_flag_equals_" + minor_version
         native.config_setting(
-            name = python_version_constraint_setting,
-            flag_values = {":python_version": python_version},
+            name = equals_minor_version_name,
+            flag_values = {":python_version": minor_version},
+        )
+
+        matches_minor_version_names = [equals_minor_version_name]
+
+        for micro_version in micro_versions:
+            is_micro_version_name = "is_python_" + micro_version
+            native.config_setting(
+                name = is_micro_version_name,
+                flag_values = {":python_version": micro_version},
+                visibility = ["//visibility:public"],
+            )
+            matches_minor_version_names.append(is_micro_version_name)
+
+        selects.config_setting_group(
+            name = "is_python_" + minor_version,
```

Comment:
@rickeylev, this will create the following targets:
```
//python/config_settings:is_python_3.10
//python/config_settings:is_python_3.10_2
//python/config_settings:is_python_3.10_3
//python/config_settings:is_python_3.10_4
//python/config_settings:is_python_3.10_5
//python/config_settings:is_python_3.10_6
//python/config_settings:is_python_3.10_7
//python/config_settings:is_python_3.10_8
```

and it is super confusing when looking at them through `bazel query`. I know that not many people might look, but having this config_setting_group with `_` prefix and then having a `native.alias` with `is_python_` prefix would work around this issue.

Reply:
Good catch. Yes, that is confusing, especially since the micro-version names are an `.N` prefix. Added the aliased config group, as suggested.